components/archiving/caseController.js

```diff
@@ -0,0 +1,439 @@
+    errors.push(
+      `Archive target '${newCase.archiveTarget}' from request does not match archive target '${existingCase.archiveTarget}' for case ${newCase.caseIdentifier}.`
+    );
+  }
+  return errors;
+};
+
+const addCaseArchive = async (req, res, next) => {
+  try {
+    //authenticate api-key from header before continuing
+    await authenticator.authenticateRequest(req);
+
+    const existingCases = await archiveDao.getByArchiveEntityIdentifier(
+      req.body.caseIdentifier,
+      false,
+      caseEntityType
+    );
+    if (existingCases == null || !existingCases.length) {
+      await upsert(req.body, true);
+      res.status(201).end();
+    } else {
+      for (let existingCase of existingCases) {
+        // check for conflicting changes
+        let errors = hasConflictingChanges(existingCase, req.body);
+        if (errors.length) {
+          await archiveDao.setEntityArchiveDoNotProcess(
+            existingCase.entityIdentifier
+          );
+          caseArchiveStopProcessing.inc({
+            caseIdentifier: existingCase.entityIdentifier,
+          });
+          for (const e of errors) {
+            logger.error(e);
+          }
+          throw new ConflictingDataError(errors.join('\n'));
+        }
+        if (!hasArchivingStarted(existingCase)) {
+          // can modify a case that hasn't been archived if there are no errors
+          await upsert(req.body, false);
+          let updatedCase = await archiveDao.getByArchiveEntityIdentifier(
+            req.body.caseIdentifier,
+            false,
+            caseEntityType
+          );
+          res.status(200).send(replaceGenericKeyName(updatedCase));
+          return true;
+        } else {
+          // if case has started archiving, can only modify case metadata
+          await archiveDao.updateMetadata(
+            req.params.caseIdentifier,
+            req.body.metadata
+          );
+          let updatedCase = await archiveDao.getByArchiveEntityIdentifier(
+            req.body.caseIdentifier,
+            false,
+            caseEntityType
+          );
+          res.status(200).send(replaceGenericKeyName(updatedCase));
+          return true;
+        }
+        /* NOTE: if we keep encountering HALP actions that don't require any further fixing other than
+           hitting the case/<caseIdentifier>/resume-archiving endpoint because the underlying discrepancy
+           gets resolved upstream, we might want to adjust these last two conditions to also clear the
+           do-not-process flag */
+      }
+    }
+  } catch (e) {
+    handleErrors(
+      e,
+      `Error adding case ${req.body.caseIdentifier}`,
+      logger,
+      next
+    );
+  }
+};
+
+const upsert = (caseInfo, createNewArchive) => {
+  return archiveDao.addArchiveEntity(
+    replaceCaseKeyName(caseInfo),
+    createNewArchive,
+    caseEntityType
+  );
+};
+
+const filesCopiedToOffsiteStagingDir = async (req, res, next) => {
```

Comment:
Unclear from the name that this function is doing something. Same with others below